### PR TITLE
Polyfills and (temp) performance fix

### PR DESCRIPTION
**Pull request by wiredearp**
@nealstewart

### CODE REVIEW

File: dist/spiritual-gui-0.2.7.js [hunk 1]
```diff
@@ -10024,6 +9931,22 @@ gui.Spirit = gui.Class.create(Object.prototype, {
 	oncrawler: function(crawler) {},
 
 
+	// Layout ....................................................................
```
Comment (nealstewart) [inline dist/spiritual-gui-0.2.7.js]: ? commented out stuff ?

Comment (wiredearp) [inline dist/spiritual-gui-0.2.7.js]: It's the next project.

**Comment by wiredearp** [comment]
> why is this important?

Now you know. That's it.

File: test/tests/specific/gui.Crawler.spec.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+	 */
+	it('should crawl the entire tree', function() {
+		new gui.Crawler().descend(this.sandbox, {
+			handleElement: function(elm) {
+				elm.setAttribute('visited', 'true');
+			}
+		});
+		var visited = this.sandbox.querySelectorAll('[visited]');
+		var allelms = this.sandbox.querySelectorAll('*');
+		expect(visited.length).toBe(allelms.length);
```
Comment (nealstewart) [inline test/tests/specific/gui.Crawler.spec.js]: optional, but it is a better idea to test that the elements in the sets are the same, than that they have the same length.

**Comment by nealstewart** [comment]
:green_apple:

**Pull request merged**